libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -43,8 +46,52 @@ type Enabled struct {
 
 type PluginConfig []map[string]common.Config
 
+func trimAfter(value string, a string) string {
+	// Get substring after a string.
+	pos := strings.LastIndex(value, a)
+	if pos == -1 {
+		return ""
+	}
+	adjustedPos := pos + len(a)
+	if adjustedPos >= len(value) {
+		return ""
+	}
+	return value[adjustedPos:len(value)]
+}
+
+func findDaemonsConfigFlag() string {
+	// Search in processes and search for config in flags
+	psRes, _ := exec.Command("bash", "-c", "ps aux | grep -- -kubeconfig=").Output()
+	psResStr := string(psRes)
+	trimmedPsResStr := trimAfter(psResStr, "--kubeconfig=")
+	arr := strings.Split(trimmedPsResStr, " ")
+	kubeConfigPath := arr[0]
+	return kubeConfigPath
+}
+
+func getSystemKubeConfig() string {
+	homeKubeConfig := os.Getenv("HOME") + "/.kube/config"
+	if _, err := os.Stat(homeKubeConfig); !os.IsNotExist(err) {
+		return homeKubeConfig
+	}
+	envKubeConfig := os.Getenv("KUBECONFIG")
+	if _, err := os.Stat(envKubeConfig); !os.IsNotExist(err) {
+		return envKubeConfig
+	}
+	kubeConfigPath := findDaemonsConfigFlag()
+	if _, err := os.Stat(kubeConfigPath); !os.IsNotExist(err) {
+		return kubeConfigPath
+	}
+	kubeletConfig := "/etc/kubernetes/kubelet.conf"
```

Comment:
I see you want to target beats running at worker nodes, not at pods. `/etc/kubernetes/kubelet.conf` is not a standard, although extensively used because of `kubeadm`. AFAIK, official kubernetes clients don't use this path, which makes me think this is an unexpected behaviour for an application.

I would remove it, if you think there is value in this, probably opening a new issue for discussing is the non blocking path for this PR.

Reply:
Ok I can open a follow-up issue for discussing any enhancements in the config discovery approach and catch as many cases as possible.